socorro/external/es/connection_context.py

```diff
@@ -196,15 +200,11 @@ def delete_expired_indices(self):
         """
         policy = datetime.timedelta(weeks=self.config.retention_policy)
         cutoff = (utc_now() - policy).replace(tzinfo=None)
+        cutoff = cutoff.strftime(self.config.elasticsearch_index)
 
         was_deleted = []
         for index_name in self.get_indices():
-            # strptime ignores week numbers if a day isn't specified, so we append
-            # '-1' and '-%w' to specify Monday as the day.
-            index_date = datetime.datetime.strptime(
-                index_name + "-1", self.config.elasticsearch_index + "-%w"
-            )
-            if index_date >= cutoff:
```

Comment:
I reworked this. We can convert the cutoff to a string and then do string comparisons and skip the whole Monday weirdness.